archetypes/se/basic/src/main/resources/src/main/java/__pkg__/Main.java.mustache

```diff
@@ -49,83 +40,58 @@ public final class Main {
-            .config(config.get("server"))
-#else
-            .routing(createRouting())
-#end
-#if( $jsonSupport.matches("y|yes|true") )
-            .addMediaSupport(JsonpSupport.create())
-#end
-            .build();
+
+        // Get webserver config from the "server" section of application.yaml
+        ServerConfiguration serverConfig =
+                ServerConfiguration.create(config.get("server"));
+
+        WebServer server = WebServer.create(serverConfig, createRouting(config));
 
         // Try to start the server. If successful, print some info and arrange to
         // print a message at shutdown. If unsuccessful, print the exception.
         server.start()
-            .thenAccept(ws -> {
-#if( $restService.matches("y|yes|true") )
-                System.out.println("WEB server is up! http://localhost:" + ws.port() + "${restServicePath}");
-#else
-                System.out.println("WEB server is up! http://localhost:" + ws.port());
-#end
-                ws.whenShutdown().thenRun(()
-                    -> System.out.println("WEB server is DOWN. Good bye!"));
+                .thenAccept(ws -> {
+                    System.out.println(
+                            "WEB server is up! http://localhost:" + ws.port() + "/greet");
+                    ws.whenShutdown().thenRun(()
+                            -> System.out.println("WEB server is DOWN. Good bye!"));
                 })
-            .exceptionally(t -> {
-                System.err.println("Startup failed: " + t.getMessage());
-                t.printStackTrace(System.err);
-                return null;
-            });
+                .exceptionally(t -> {
+                    System.err.println("Startup failed: " + t.getMessage());
+                    t.printStackTrace(System.err);
+                    return null;
+                });
 
         // Server threads are not daemon. No need to block. Just react.
+
         return server;
     }
 
     /**
      * Creates new {@link Routing}.
      *
-     * @return configured routing
+     * @return routing configured with JSON support, a health check, and a service
      * @param config configuration of this server
      */
-#if( $applicationYaml.matches("y|yes|true") )
     private static Routing createRouting(Config config) {
-#else
-    private static Routing createRouting() {
-#end
-#if( $metricsSupport.matches("y|yes|true") )
         MetricsSupport metrics = MetricsSupport.create();
-#end
-#if( $healthSupport.matches("y|yes|true") )
+        GreetService greetService = new GreetService(config);
         HealthSupport health = HealthSupport.builder()
-                .add(HealthChecks.healthChecks())   // Adds a convenient set of checks
+                .addLiveness(HealthChecks.healthChecks())   // Adds a convenient set of checks
                 .build();
-#end
 
         return Routing.builder()
-#if( $healthSupport.matches("y|yes|true") )
+                .register(JsonSupport.create())
```

Comment:
This is wrong place to register this.
Please switch to new media support
Should be registered on `Server.builder().addMediaSupport(JsonpSupport.create())`